src/provider/controller/IntentHandler.ts

```diff
@@ -86,34 +89,51 @@ export class IntentHandler {
             // Resolve intent immediately
             return this.fireIntent(intent, apps[0]);
         } else {
-            console.log(`${apps.length} apps found to resolve intent '${intent.type}', showing resolver'`);
+            console.log(`${apps.length} apps found to resolve intent '${intent.type}', delegating app choice'`);
 
-            return this.queueResolve(intent, apps);
+            return handleAppChoice(intent, apps);
         }
     }
 
     private async queueResolve(intent: Intent, applications: Application[]): Promise<IntentResolution> {
         if (this._resolvePromise) {
             console.log(`Resolver showing, re-resolving intent '${intent.type}' when resolver closes'`);
 
-            this._resolvePromise = this._resolvePromise.catch(() => {}).then(() => this.startResolve(intent));
-
-            return this._resolvePromise;
+            this._resolvePromise = this._resolvePromise.catch(() => {}).then(() => this.startResolve(intent, this.showResolver.bind(this)));
```

Comment:
Don't we have an `allowReject` async util somewhere to avoid the `catch(()=>{})`?

Reply:
We do but it's not quite the same
`allowReject(promise)` - a promise that resolves or rejects when the promise resolves or rejects respectively (but rejection won't print an error to the console)
`promise.catch(() => {})` - a promise that resolves when the promise either resolves or rejects

Reply:
Hmmm, ok, that makes sense.